app/frontend/src/utils/hooks.ts

```diff
@@ -39,4 +41,63 @@ function useSafeState<State>(
   return [state, safeSetState];
 }
 
-export { useIsMounted, useSafeState };
+export interface GeocodeResult {
+  name: string;
+  simplifiedName: string;
+  location: LngLat;
+}
+
+const NOMINATIM_URL = process.env.REACT_APP_NOMINATIM_URL;
+
+const useGeocodeQuery = () => {
+  const [isLoading, setIsLoading] = useState(false);
+  const [error, setError] = useState<string | undefined>(undefined);
+  const [results, setResults] = useState<GeocodeResult[] | undefined>([]);
+
+  const query = useCallback(async (value: string) => {
+    if (!value) {
+      return;
+    }
+    setIsLoading(true);
+    setError(undefined);
+    setResults(undefined);
+    const url = `${NOMINATIM_URL!}search?format=jsonv2&q=${encodeURIComponent(
+      value
+    )}&addressdetails=1`;
+    const fetchOptions = {
+      headers: {
+        Accept: "application/json",
+        "Content-Type": "application/json;charset=UTF-8",
+      },
+      method: "GET",
+    };
+    try {
+      const response = await fetch(url, fetchOptions);
+      const nominatimResults = (await response.json()) as Array<NominatimPlace>;
```

Comment:
```suggestion
      const nominatimResults: NominatimPlace[] = await response.json();
```
Nitpick: Personal preference to not use type assertions if possible, but probably doesn't make a big difference in this specific case since the type is loose/bad anyway...